src/librustdoc/clean/types.rs

```diff
@@ -746,34 +746,23 @@ impl Item {
         Some(tcx.visibility(def_id))
     }
 
-    fn attributes_without_repr(&self, tcx: TyCtxt<'_>, is_json: bool) -> Vec<String> {
+    /// Get a list of attributes excluding `#[repr]`.
+    ///
+    /// Only used by the HTML output-format.
+    fn attributes_without_repr(&self, tcx: TyCtxt<'_>) -> Vec<String> {
         const ALLOWED_ATTRIBUTES: &[Symbol] =
-            &[sym::export_name, sym::link_section, sym::no_mangle, sym::non_exhaustive];
+            &[sym::export_name, sym::link_section, sym::non_exhaustive];
         self.attrs
             .other_attrs
             .iter()
             .filter_map(|attr| {
-                // NoMangle is special cased, as it appears in HTML output, and we want to show it in source form, not HIR printing.
-                // It is also used by cargo-semver-checks.
-                if matches!(attr, hir::Attribute::Parsed(AttributeKind::NoMangle(..))) {
+                // NoMangle is special cased, as it's currently the only structured
+                // attribute that we want to appear in HTML output.
```

Comment:
As a separate concern unrelated to this PR, perhaps HTML output should list both `#[no_mangle]` and `#[export_name]` in their edition 2024 form i.e. wrapped in `unsafe`?

Reply:
Please open an issue then so it's not forgotten.